### PR TITLE
chunk: "setMutRowJSON" should allocate one more byte for null bitmap

### LINKED ISSUE #5601: MutRow.setDatum paniced
## 1. What did you do?

```sql
drop table if exists t;
create table t(a json, b bigint);

begin;
insert into t values("[1]", 1);
select * from t
commit;
```

## 2. What did you expect to see?

```txt
+------+------+
| a    | b    |
+------+------+
| [1]  |    1 |
+------+------+
1 row in set (0.11 sec)
```

## 3. What did you see instead?

```txt
2018/01/10 14:04:00.528 conn.go:410: [error] lastCmd select * from t, runtime error: index out of range, goroutine 125 [running]:
github.com/pingcap/tidb/server.(*clientConn).Run.func1(0xc4202d3a00, 0xc420867e0f)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/conn.go:408 +0x10e
panic(0x1cd1e60, 0x285a480)
        /Users/jianzhang.zj/opt/go-1.9/src/runtime/panic.go:491 +0x283
github.com/pingcap/tidb/util/chunk.MutRow.SetDatum(0xc42034a200, 0x0, 0x0, 0x12, 0x3, 0xc420416ba6, 0x15, 0x1b, 0x0, 0x0)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/util/chunk/mutrow.go:305 +0xee3
github.com/pingcap/tidb/util/chunk.MutRow.SetDatums(0xc42034a200, 0x0, 0xc4207b44d0, 0x3, 0x3)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/util/chunk/mutrow.go:270 +0xe9
github.com/pingcap/tidb/executor.(*UnionScanExec).NextChunk(0xc4208a4500, 0x2ed2588, 0xc42041f340, 0xc42034a080, 0xc42053ddf0, 0x0)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/executor/union_scan.go:128 +0xd8
github.com/pingcap/tidb/executor.(*ProjectionExec).NextChunk(0xc420eb9380, 0x2ed2588, 0xc42041f340, 0xc42034a180, 0x2, 0x2)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/executor/executor.go:637 +0x83
github.com/pingcap/tidb/executor.(*recordSet).NextChunk(0xc420eb78c0, 0x2ed2588, 0xc42041f340, 0xc42034a180, 0xc420eb78c0, 0xc42034a180)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/executor/adapter.go:103 +0x56
github.com/pingcap/tidb/server.(*tidbResultSet).NextChunk(0xc420ec5560, 0x2ed2588, 0xc42041f340, 0xc42034a180, 0x4, 0x400)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/driver_tidb.go:302 +0x51
github.com/pingcap/tidb/server.(*clientConn).writeChunks(0xc4202d3a00, 0x2ed2588, 0xc42041f340, 0x264ba60, 0xc420ec5560, 0x0, 0xc420ec5560, 0x1)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/conn.go:936 +0x3ef
github.com/pingcap/tidb/server.(*clientConn).writeResultset(0xc4202d3a00, 0x2ed2588, 0xc42041f340, 0x264ba60, 0xc420ec5560, 0xc420530000, 0x0, 0x0)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/conn.go:861 +0x668
github.com/pingcap/tidb/server.(*clientConn).handleQuery(0xc4202d3a00, 0x2ed2588, 0xc42041f340, 0xc4203a3f51, 0xf, 0x0, 0x0)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/conn.go:810 +0x115
github.com/pingcap/tidb/server.(*clientConn).dispatch(0xc4202d3a00, 0xc4203a3f51, 0x10, 0x10, 0x0, 0x0)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/conn.go:592 +0x595
github.com/pingcap/tidb/server.(*clientConn).Run(0xc4202d3a00)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/conn.go:452 +0x1b8
github.com/pingcap/tidb/server.(*Server).onConn(0xc42005b320, 0x264fa20, 0xc4202ce320)
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/server.go:316 +0x144
created by github.com/pingcap/tidb/server.(*Server).Run
        /Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/server/server.go:262 +0x496
```

## 4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```sh
$./bin/tidb-server -V
Release Version: v1.1.0-alpha-401-gb7e7cd5d9
Git Commit Hash: b7e7cd5d9a0c38fe248e45bf1e89fe1e3f904aed
Git Commit Branch: master
UTC Build Time:  2018-01-10 05:29:09
```

### PR DESCRIPTION
**Pull request by zz-jason**
fix #5601

### CODE REVIEW

**Comment by zz-jason** [comment]
@coocood @XuHuaiyu PTAL

File: util/chunk/mutrow.go [hunk 1]
```diff
@@ -336,7 +336,7 @@ func setMutRowJSON(col *column, j json.BinaryJSON) {
 	if len(col.data) >= dataLen {
 		col.data = col.data[:dataLen]
 	} else {
-		buf := make([]byte, dataLen)
+		buf := make([]byte, dataLen+1)
```
Comment (shenli) [inline util/chunk/mutrow.go]: Add comment for it.

**Comment by coocood** [comment]
LGTM

**Pull request merged**